modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/AbstractEventEndpoint.java

```diff
@@ -298,6 +299,47 @@ public void activate(ComponentContext cc) {
 
   }
 
+  /* As the list of event ids can grow large, we use a POST request to avoid problems with too large query strings */
+  @POST
+  @Path("workflowProperties")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "workflowProperties", description = "Returns workflow properties for the specified events",
+             returnDescription = "The workflow properties for every event as JSON", restParameters = {
+                @RestParameter(name = "eventIds", description = "A JSON array of ids of the events", isRequired = true, type = RestParameter.Type.STRING)},
+             reponses = {
+                @RestResponse(description = "Returns the workflow properties for the events as JSON", responseCode = HttpServletResponse.SC_OK),
+                @RestResponse(description = "The list of ids could not be parsed into a json list.", responseCode = HttpServletResponse.SC_BAD_REQUEST)
+              })
+  public Response getEventWorkflowProperties(@FormParam("eventIds") String eventIds) throws UnauthorizedException {
+    if (StringUtils.isBlank(eventIds)) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+
+    JSONParser parser = new JSONParser();
+    List<String> ids;
+    try {
+      ids = (List<String>) parser.parse(eventIds);
+    } catch (org.json.simple.parser.ParseException e) {
+      logger.error("Unable to parse '{}' because: {}", eventIds, ExceptionUtils.getStackTrace(e));
```

Comment:
I can change this, but is it really wise? This confuses the hell out of Opencast users (because not even developers can really read exception traces) and the developer doesn't get a chance to debug the code - unless the user seeing it reports it as is. I do see the opportunity for spamming, though...

Reply:
@lkiesow 

Please note that at least in our case the "client" (technically the Admin UI in this case) is in general used by a user which always never has any software development skills and sometimes not even a computer science background. That user is in general neither capable nor happy to help us. If absolutely required, we can call the user and ask him to perform the task that fails so that we can follow live in our logs but we usually try to avoid even this as we don't want to bother users with our problems.

From that perspective, the argument that such problems should be reported to the client and not logged in Opencast at all isn't very helpful as we in general don't have access to whatever is reported to the client - and displaying technical error messages like this one in a user-facing component seems a total usability NO-GO to me.

That is exactly the kind of things we need to be able to see in our logs to be able to debug problems without having access to the client. Note that we in general also don't have access to other clients (e.g. learning management systems).

BTW: In this particular case, the log level ERROR even seems perfectly appropriate to me: As this endpoint is internal, the only official client using it is the Admin UI. But if the Admin UI sends a list that cannot be parsed, this indeed is a software bug in Opencast which needs to be fixed. Why shouldn't this be logged as error then?

Or to put it the other way around:

Assuming that...

- ... displaying technical error messages (stack trace, cannot parse JSON, ...) in a user-facing component is a usability NO-GO. The UI would say something like "Operation failed", but not "Operation failed because media package xY could not be parsed"
- ... we in general don't have access to what is reported to the client

... how would we then debug a problem occurring in the client that is reported by a user?

Memo: Once I've got plenty of time, I should bring up a discussion on list about what people are doing with the log and what they expects to be logged. Once I've stranded on an isle, I shall write a log level guide ;-)